### PR TITLE
Add support for a dynamic number of squads in SSdirection

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Add support for a dynamic number of squads in SSdirection

## Why It's Good For The Game

This is some initial work for working with an unlimited number of factions / squads.

### CODE REVIEW

**Comment by psykzz** [comment]
PR is still a work in progress.

**Comment by psykzz** [comment]
Good to go now, find out defining an arg as `proc(/datum/full/path, arg2, arg3)` no only fails silently but also blocks further args.

File: code/controllers/stonedmc/subsystem/direction.dm [hunk 1]
```diff
@@ -4,8 +4,8 @@ SUBSYSTEM_DEF(direction)
 	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
 	wait = 1 SECONDS
 
-	// this is a map of defines to mob references, eg; list(CHARLIE_SL = <mob ref>, XENO_NORMAL_QUEEN = <mob ref>)
-	var/list/leader_mapping = list()
+	// this is a map of defines to mob references, eg; list(FACTION_ID = <mob ref>, FACTION_ID2 = <mob ref>)
+	var/list/leader_mapping[0]
```
Comment (spookydonut) [inline code/controllers/stonedmc/subsystem/direction.dm]:
```suggestion
	var/list/leader_mapping = list()
```

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 2]
```diff
@@ -35,6 +35,7 @@
 	for(var/upgrade in GLOB.xenoupgradetiers)
 		xenos_by_upgrade[upgrade] = list()
 
+
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm]: don't commit whitespace changes

**Pull request merged**